packages/expo-application/ios/EXApplication/EXApplication.m

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+#import <UMCore/UMUtilities.h>
+#import <EXApplication/EXApplication.h>
+#import <UIKit/UIKit.h>
+
+@interface EXApplication ()
+
+
+@end
+
+@implementation EXApplication
+
+UM_EXPORT_MODULE(ExpoApplication);
+
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+UM_EXPORT_METHOD_AS(getIosIdForVendorAsync, getIosIdForVendorAsyncWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject)
+{
+  if(NSClassFromString(@"UIDevice") && [UIDevice instancesRespondToSelector:@selector(identifierForVendor)]) {
+    resolve([[UIDevice currentDevice].identifierForVendor UUIDString]);
+  }
+  else{
+    resolve([NSNull null]);
+  }
+}
+
+UM_EXPORT_METHOD_AS(getInstallationTimeAsync, getInstallationTimeAsyncWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject)
+{
+  NSURL* urlToDocumentsFolder = [[[NSFileManager defaultManager] URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask] lastObject];
+  NSError *error = nil;
+  NSDate *installDate = [[[NSFileManager defaultManager] attributesOfItemAtPath:urlToDocumentsFolder.path error:&error] objectForKey:NSFileCreationDate];
+  if(error){
+    reject(@"ERR_APPLICATION", @"Unable to get installation time of this application.", error);
+  }
+  else{
+    NSTimeInterval timeInMiliseconds = [installDate timeIntervalSince1970] * 1000;
+    NSNumber* timeInterval = @(timeInMiliseconds);
```

Comment:
c-style instead of c++ style, also "timeInterval" is a confusing name esp when this is not an NSTimeInterval
```suggestion
    NSNumber *timeNumber = @(timeInMilliseconds);
```